h/views/activity.py

```diff
@@ -95,22 +95,17 @@ def user_link(userid):
 class GroupSearchController(SearchController):
     """View callables unique to the "activity.group_search" route."""
 
+    def __init__(self, group, request):
+        super(GroupSearchController, self).__init__(request)
+        self.group = group
+
     @view_config(request_method='GET')
     def search(self):
         result = super(GroupSearchController, self).search()
 
-        pubid = self.request.matchdict['pubid']
-        result['opts'] = {'search_groupname': pubid}
-
-        try:
-            group = (self.request.db.query(models.Group)
-                     .filter_by(pubid=pubid).one())
-        except exc.NoResultFound:
-            return result
-
-        result['opts']['search_groupname'] = group.name
```

Comment:
No longer need to fetch the `Group` from the db itself, or worry about when the group doesn't exist